### ISSUE TITLE
Implement standardized error handling across all extractors

### ISSUE THREAD

**Issue by PulkitMishra** [tensorlakeai/indexify-extractors#214]
Currently, the extractor framework lacks a consistent approach to error handling
- Some extractors use try-except blocks with their own Exceptions, some with generic Exceptions, while others don't use error handling at all
- Error messages are inconsistent in format and detail level.
- Some errors are silently caught and logged, while others are raised and propagate etc

This could lead to several issues in the future and to avoid the same its best to have a standardized way of handling errors across the lib. 

To begin with we could have few custom exception classes and implement error translation for third-party libraries. Behaviour of logger can also be changed accordingly

**Comment by diptanu** [comment]
@PulkitMishra Just seeing this! I love your idea. In most cases the errors are not caught and crashes the entire executor and fails silently. Feel free to contribute some PRs related to this.

**Comment by PulkitMishra** [comment]
cool @diptanu . i only went through the docs for a few minutes so correct me if i am wrong but my understanding is that input raw data -> extraction graph -> output content, embeddings, metadata. here, extraction graphs are a combination of extraction policies and extraction policies themselves are sort of instances of extractors (ex - moondream is an extractor but moondream with a modified prompt is a policy).

so i thought having consistency in raising errors here in extractors would make sense given they are being used upstream.

will code something up and send a draft PR in an hour or so. have a look at it and let me know if the general idea of implementation makes sense

**Comment by diptanu** [comment]
@PulkitMishra Yes, that's right! If you have thoughts around improving the docs please let us know :)
